[add] DB 수정/삭제 기능 추가 및 TC, ModifyCommand 기능 추가

happy-dev-travel opened the pull request:
- for modifyCommand
  - 기능이 구현되었습니다. PR이 열려있는 상태에서 하는 바람에 커밋이 합쳐졌네요..ㅠ
  - 버그가 발견되어 추가하였습니다

- for DB class
   - modifyEmployee, deleteEmployee가 추가되었습니다
   - modifyEmployee는 변경되는 Employee를 hash에서 삭제하고 새로 추가합니다. 컬럼의 값 별로 hashing이 되어있다보니.. 수정한 값이 새롭게 hashing할 필요가 있을 수 있어서 삭제 후 추가함. 지금은 컬럼 hash별로 모두 삭제, 모두 추가하고 있는데.. 요구사항은 하나의 컬럼만 변경되니.. 변경되는 부분만 처리하도록 할 필요도 있을 것 같아요 의견 부탁드립니다

그리고 DB를 추상화해서 리스트 기반의 EmployeeDB를 만들필요는 없을까 고민이 되네요

happy-dev-travel committed:
[add] DB 수정/삭제 기능 추가 및 TC
- modifyEmployee, deleteEmployee가 추가되었습니다
- modifyEmployee는 변경되는 Employee를 hash에서 삭제하고 새로 추가합니다. 컬럼의 값 별로 hashing이 되어있다보니.. 수정한 값이 새롭게 hashing할 필요가 있을 수 있어서 삭제 후 추가함. 지금은 컬럼 hash별로 모두 삭제, 모두 추가하고 있는데.. 요구사항은 하나의 컬럼만 변경되니.. 변경되는 부분만 처리하도록 할 필요도 있을 것 같아요 의견 부탁드립니다

그리고 DB를 추상화해서 리스트 기반의 EmployeeDB를 만들필요는 없을까 고민이 되네요

smilesw6448 commented on src/main/java/database/EmployeeDB.java:
```diff
@@ -109,20 +61,62 @@ public Employee deleteEmployee(Employee deleteEmployee){
+        return backup; //수정 전 상태를 반환
+    }
+
+    private void AddToColumnHash(String columnName,String value,Employee employeeToAdd){
```
함수명 통일하셨군요! 가독성 훨씬 좋아졌습니다

happy-dev-travel committed:
Merge remote-tracking branch 'origin/main' into dev_이동길

happy-dev-travel committed:
[add] ModifyCommand 기능 추가
- ModifyCommand를 구현하였습니다

jw451 commented on src/main/java/command/mod/ModifyCommand.java:
```diff
@@ -15,7 +15,52 @@ public ModifyCommand(CommandExecutor schCommand) {
+    }
+
+    private List<Employee> modify(EmployeeDB db, List<Employee> employees) {
+        List<Employee> beforeModifiedEmployees = new ArrayList<>();
```
beforeModifiedEmployees로 변수명을 지으신 이유가 있을까요?

happy-dev-travel replied:
요구사항에 보니 수정을 했을 때의 출력되는 내용은 수정 전 데이터라고 되어 있어서 그렇습니다

jw451 reviewed (approved):
고생하셨습니다.
전반적으로 깔끔하고, 이상없어 보입니다.

jw451 commented on src/main/java/command/mod/ModifyCommand.java:
```diff
@@ -15,7 +15,52 @@ public ModifyCommand(CommandExecutor schCommand) {
+    }
+
+    private List<Employee> searchEmployees(UserRequest request, EmployeeDB db) throws Exception {
+        request.setCommand(COMMAND.SCH);
```
request의 기존COMMAND가 MOD였는데 SCH로 바뀌게되어도 괜찮은지요?

happy-dev-travel replied:
앗! 리뷰 감사합니다!! SearchCommand를 사용해야 해서 변경을 했는데, 이대로 진행하면 나중에 프린트 부분에서 Command가 잘못 출력되겠네요... 수정하겠습니다!! 감사합니다

bw2218 replied:
혹시 request에 setCommand하여 SCH로 바꾸는 방법이 아닌 어떤 방안으로 수정하실지 미리 알 수 있을까요.....? ^^

happy-dev-travel committed:
[bug] 커맨드 변경으로 인한 출력 버그 수정
- 지원님 리뷰로 발견되어 수정합니다
- 내부에서 커맨드를 SCH로 변경하고 되돌리지 않았네요 이대로면 출력할 때 MOD가 전부 SCH로 나오게 됩니다. 리뷰 감사합니다!
- UserRequest를 변경하지 말고 새로운 객체를 만드는게 나을 수도 있겠는데.. 딱 커맨드 종류만 다르니 일단은 set 함수를 사용하였습니다

bw2218 commented on src/main/java/command/mod/ModifyCommand.java:
```diff
@@ -15,7 +15,51 @@ public ModifyCommand(CommandExecutor schCommand) {
+
+    private List<Employee> searchEmployees(UserRequest request, EmployeeDB db) throws Exception {
+        request.setCommand(COMMAND.SCH);
+        return this.schCommand.run(request, db);
```
기존 함수 재활용 하는부분 코드가 깔끔한것 같습니다.....^^
